File: Java examples/src/frontController/Dispatcher.java
package frontController;

class Dispatcher {
    private HomeController homeController;
    private ErrorController errorController;

    public Dispatcher() {
        homeController = new HomeController();
        errorController = new ErrorController();
    }

    public void dispatch(String request) {
        if ("HOME".equalsIgnoreCase(request)) {
            homeController.processRequest();
        } else {
            errorController.processRequest();
        }
    }
}

File: C# examples/FrontControllerPattern/Dispatcher.cs
namespace FrontControllerPattern
{
    public class Dispatcher
    {
        private readonly HomeController _homeController;
        private readonly ErrorController _errorController;

        public Dispatcher()
        {
            _homeController = new HomeController();
            _errorController = new ErrorController();
        }

        public void Dispatch(string request)
        {
            if (request.Equals("HOME", StringComparison.OrdinalIgnoreCase))
            {
                _homeController.ProcessRequest();
            }
            else
            {
                _errorController.ProcessRequest();
            }
        }
    }
}
